apps/ledger-live-desktop/src/renderer/components/SignMessageConfirm/index.jsx

```diff
@@ -55,43 +59,68 @@ type Props = {
   signMessageRequested: TypedMessageData | MessageData,
 };
 
-const SignMessageConfirm = ({ device, account, signMessageRequested: message }: Props) => {
+const SignMessageConfirm = ({ device, account, signMessageRequested }: Props) => {
   const type = useTheme("colors.palette.type");
   const { t } = useTranslation();
 
   if (!device) return null;
+  const inferredNanoFields = (() => {
+    try {
+      if (account.currency.family === "ethereum") {
+        const parsedMessage =
+          typeof signMessageRequested.message === "string"
+            ? JSON.parse(signMessageRequested.message)
+            : signMessageRequested.message;
 
-  const fields = [];
+        return isEIP712Message(signMessageRequested.message)
+          ? getNanoDisplayedInfosFor712(parsedMessage)
+          : null;
+      }
+      throw new Error();
```

Comment:
why throw an errror here as it'll be catched and then return null ?

Reply:
You're right here I could return `null` directly, I'm just replicating the logic found in `apps/ledger-live-mobile/src/screens/SignMessage/01-Summary.tsx`.

Doesn't change anything here, but I can change it if you want 👌